lib/mongoose_push/service/apns.ex

```diff
@@ -5,77 +5,64 @@ defmodule MongoosePush.Service.APNS do
 
   @behaviour MongoosePush.Service
   require Logger
-  alias Pigeon.APNS
-  alias Pigeon.APNS.Config
-  alias Pigeon.APNS.Notification
+  alias Sparrow.APNS
+  alias Sparrow.APNS.Notification
+  alias MongoosePush.Application
   alias MongoosePush.Service
-  alias MongoosePush.Pools
   alias MongoosePush.Service.APNS.Certificate
+  alias MongoosePush.Service.APNS.State
 
   @priority_mapping %{normal: "5", high: "10"}
 
-  @spec prepare_notification(String.t(), MongoosePush.request()) ::
+  @spec prepare_notification(String.t(), MongoosePush.request(), atom()) ::
           Service.notification()
-  def prepare_notification(device_id, %{alert: nil} = request) do
+  def prepare_notification(device_id, %{alert: nil} = request, _pool) do
     # Setup silent notification
-    %{"content-available" => 1}
-    |> Notification.new(device_id, request[:topic], request[:data])
+    Notification.new(device_id, Map.get(request, :mode, :prod))
+    |> Notification.add_content_available(1)
+    |> maybe(:add_apns_topic, request[:topic])
+    |> add_data(request[:data])
     |> maybe_mutable_content(request[:mutable_content])
-    |> Notification.put_priority(@priority_mapping[request[:priority]])
+    |> maybe(:add_apns_priority, @priority_mapping[request[:priority]])
   end
 
-  def prepare_notification(device_id, request) do
+  def prepare_notification(device_id, request, pool) do
     # Setup non-silent notification
     alert = request.alert
+    default_topic = State.get_default_topic(pool)
 
-    %{
-      "alert" => %{
-        "title" => alert.title,
-        "body" => alert.body
-      },
-      "badge" => alert[:badge],
-      "category" => alert[:click_action],
-      "sound" => alert[:sound]
-    }
-    |> Notification.new(device_id, request[:topic], request[:data])
+    Notification.new(device_id, Map.get(request, :mode, :prod))
+    |> maybe(:add_apns_topic, request[:topic] || default_topic)
+    |> add_data(request[:data])
     |> maybe_mutable_content(request[:mutable_content])
-    |> Notification.put_priority(@priority_mapping[request[:priority]])
+    |> maybe(:add_apns_priority, @priority_mapping[request[:priority]])
+    |> Notification.add_title(alert.title)
```

Comment:
We should group `maybe` calls together, and push all "required" direct calls to the top.